snap-gpf/src/main/java/org/esa/snap/core/gpf/internal/OperatorContext.java

```diff
@@ -1286,10 +1287,19 @@ public void setRequiresAllBands(boolean requiresAllBands) {
     }
 
     public synchronized void executeOperator(ProgressMonitor pm) {
-        if (!executed) {
+        if (!executed.get()) {
```

Comment:
here also two new subprogress monitors, and a try-finally surround the code and at the beginning pm.beginTask(). See my change for GraphProcessor.
Currently, the pm might be started twice. In operatorContext.executeOperator() and in doExecute(). Could lead to strange behaviour of progress indicator.

Reply:
This has led to yet more changes ... The pm was actually already started twice, once for the tile computation and once for the node executions. I have harmonized this (and also changed the operator's default and the writeop's implementation of doExecute).

Reply:
But don't we need two new SubPMs here?
One at L1296 and one at L1300?
Maybe we talk later.